### PR TITLE
fix: remove obsolete allocateNodeCidrs ARM variable

### PR DESCRIPTION
**Pull request by jackfrancis**
We evaluate “if AzureCNI” directly

**Reason for Change**:

Identified an ARM variable we are unnecessarily defining in our templates.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/engine/armvariables.go [hunk 1]
```diff
@@ -299,13 +298,6 @@ func getK8sMasterVars(cs *api.ContainerService) map[string]interface{} {
 		masterVars["storageAccountBaseName"] = ""
 	}
 
-	//TODO: This could be a bug
-	if isAzureCNI {
-		masterVars["allocateNodeCidrs"] = false
-	} else {
-		masterVars["allocateNodeCidrs"] = true
```
Comment (serbrech) [inline pkg/engine/armvariables.go]: that's the no_template equivalent?

Comment (tariq1890) [reply pkg/engine/armvariables.go]: yes

**Pull request merged**